### DISCUSSION TITLE
Confusing installation instructions

### DISCUSSION THREAD

**Discussion by bound-variable** [kakoune-lsp/kakoune-lsp#601] [Q&A] [answered]
I'm confused about how to get this to work.

Are the following steps correct?

1. Install the language server protocol _client_
2. Install the relevant language server protocol _servers_ 
2.1 Some servers have built-in support, some don't
2.1.1 For those servers that _have_ built-in support, nothing further needs doing
2.1.2 For those servers that _lack_ built-in support, proceed to configure those language servers in `kak-lsp/kak-lsp.toml`
3. In case configurations to `kak-lsp/kak-lsp.toml` were done, then add the following to `kakrc`: `eval %sh{kak-lsp --kakoune -s $kak_session}  # Not needed if you load it with plug.kak.
lsp-enable`

If those steps are wrong, please explain to me the problem.

Regardless if those steps are right or not, the README would do well with some added clarity. I'd be willing to help with that.

Thanks

**Comment by krobelus** [comment, marked as answer]
your steps are correct except that step 3 is always required.
`kak-lsp --kakoune` defines the lsp-* Kakoune commands, and running `lsp-enable` (or `lsp-enable-window`) actually activates LSP features

I agree that the README is confusing and intimidating, I'm sure there are ways to improve it

**Reply by bound-variable** [reply]
Thanks.

Regarding the specific issue of installation, perhaps the largest source of confusion comes from this statement under [Usage](https://github.com/kak-lsp/kak-lsp#usage):

> To enable LSP support for configured languages (see the next section) just add the following commands to your kakrc.

Which languages we take to be _configured_ is ambiguous here. The statement directs us to the next section which discusses the configuration file. But the default settings in that file don't necessarily need any changing or _configuring_. In other words, some languages need manual configuring and some don't. But all languages need configuring of some sort. So the above statement is ambiguous between the two.

We could improve it by simply stating,

"_LSP support must be enabled by adding the following commands to your kakrc..._"

Even better, let me go through the whole README and file a pull request with welcome emendations.

**Reply by bound-variable** [reply]
Or, perhaps I can make a wiki page with a simplified set of installation instructions.

**Reply by krobelus** [reply]
Sounds good, PR is always appreciated. I see the ambiguity of "configured languages" now.
Maybe that statement should be even more concise `Add the following commands to your kakrc`.

Looking at the bigger picture, we should probably add a ToC for better overview of the steps.

**Reply by krobelus** [reply]
a wiki page would work of course but the README is the canonical source of information, always good to improve that

**Reply by bound-variable** [reply]
> Maybe that statement should be even more concise

Yes, concision is a virtue. I think the README would be better served with a restructuring. For example, the statement about `kakrc` is under the **Usage** section, but I feel it would be clearer if it were under the **Installation** section. And the same goes for the section [Language servers](https://github.com/kak-lsp/kak-lsp#language-servers) because its formatting suggests it's its own major section, but really it should be a subsection of **Installation**.

**Reply by bound-variable** [reply]
> a wiki page would work of course but the README is the canonical source of information, always good to improve that

My skills lie with English (and other academic stuff), not with computer science. Accordingly, I am more likely to make unwanted changes to a technical document like the README than I am to a more limited document such as a self-contained wiki page covering installation only.

But I'm willing to give the README a shot if your're willing to review my changes in the PR.

**Reply by krobelus** [reply]
I have moved essential configuration from Usage to a separate section, see https://github.com/kak-lsp/kak-lsp/tree/docs
